ipn/ipnserver/server.go

```diff
@@ -458,6 +458,43 @@ func (s *Server) localAPIPermissions(ci connIdentity) (read, write bool) {
 	return false, false
 }
 
+var (
+	uidMapMu sync.Mutex
+	uidMap   = map[string]string{} // "caddy" => "998"
+)
+
+// userIDFromString maps from either a numeric user id in string form
+// ("998") or username ("caddy") to its string userid ("998").
+// It returns the empty string on error.
+func userIDFromString(v string) string {
+	if v == "" || isAllDigit(v) {
+		return v
+	}
+	uidMapMu.Lock()
+	defer uidMapMu.Unlock()
+	if id, ok := uidMap[v]; ok {
+		return id
+	}
+	u, err := user.Lookup(v)
```

Comment:
I'm trying to think through whether there are cases for a long-running tailscaled where an older cached value would be wrong (since we have no way of knowing to clear it here). I guess if someone deletes a user from /etc/passwd, or if LDAP is in use, a UID could change. It doesn't seem harmful though.

Reply:
Actually I'm not sure why I even added a cache. It's pointless. Removed.